ocpmodels/trainers/energy_trainer.py

```diff
@@ -206,10 +218,15 @@ def predict(self, loader, results_file=None, disable_tqdm=False):
                 out["energy"] = self.normalizers["target"].denorm(
                     out["energy"]
                 )
-            predictions["id"].extend([str(i) for i in batch[0].sid.tolist()])
+
             predictions["energy"].extend(out["energy"].tolist())
+            if per_image:
+                predictions["id"].extend(
+                    [str(i) for i in batch[0].sid.tolist()]
+                )
+                predictions["energy"].extend(out["energy"].tolist())
+                self.save_results(predictions, results_file, keys=["energy"])
```

Comment:
you need to pull this out of this block and place it in line 212 (where it originally was)